osu.Game/Graphics/UserInterface/BreadcrumbControl.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) 2007-2017 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using System.Linq;
+using OpenTK;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.UserInterface;
+
+namespace osu.Game.Graphics.UserInterface
+{
+    public class BreadcrumbControl<T> : OsuTabControl<T>
+    {
+        public static readonly float padding = 10;
+
+        protected override TabItem<T> CreateTabItem(T value) => new BreadcrumbTabItem(value);
+
+        public BreadcrumbControl()
+        {
+            Height = 28;
+            TabContainer.Spacing = new Vector2(padding, 0f);
+            Current.ValueChanged += tab =>
+            {
+                foreach (TabItem<T> t in TabContainer.Children)
+                {
+                    if (!(t is BreadcrumbTabItem)) return;

```

Comment:
then cast it like `((BreadcrumbTabItem)t)`. That cast should throw an exception if t isn't a BreadcrumbItem